### PR TITLE
demo for modal refactor

### PR DESCRIPTION
**Pull request by EdmundMai**

### CODE REVIEW

File: src/components/file-manager/recovery-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+interface RecoveryModalState {
+  isOpen;
+  opacity;
+}
+
+class RecoveryModal extends Component<RecoveryModalProps, RecoveryModalState> {
+  state = {
+    isOpen: true,
+    opacity: 1
```
Comment (EdmundMai) [inline src/components/file-manager/recovery-modal.tsx]: you don't need this, isOpen will hide / open it for you

File: src/components/file-manager/recovery-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+interface RecoveryModalProps {}
+
+interface RecoveryModalState {
+  isOpen;
+  opacity;
+}
+
+class RecoveryModal extends Component<RecoveryModalProps, RecoveryModalState> {
+  state = {
```
Comment (EdmundMai) [inline src/components/file-manager/recovery-modal.tsx]: take a look at this, we should use the new react API for state: https://reactjs.org/docs/hooks-state.html

look at the `useState` hook

File: src/components/file-manager/recovery-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+  toggleModal (e) {
+    this.setState({ isOpen: !this.state.isOpen });
+  }
+
+  afterOpen () {
+    setTimeout(() => {
+      this.setState({ opacity: 1 });
+    });
+  }
```
Comment (EdmundMai) [inline src/components/file-manager/recovery-modal.tsx]: remove

File: src/components/file-manager/recovery-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+  beforeClose () {
+    return new Promise(resolve => {
+      this.setState({ opacity: 0 });
+      setTimeout(resolve, 200);
+    });
+  }
```
Comment (EdmundMai) [inline src/components/file-manager/recovery-modal.tsx]: remove

File: src/components/file-manager/recovery-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+              <Button onClick={e => this.toggleModal(e)}>Close window</Button>
+            </ButtonWrapper>
+          </StyledModal>
+        </ModalProvider>
+      </ThemeProvider>
+    );
+  }
+}
+
+export default RecoveryModal;
```
Comment (EdmundMai) [inline src/components/file-manager/recovery-modal.tsx]: you are not importing this to the file-manager slide right now

File: src/components/file-manager/recovery-modal.tsx (hunk 1 above)
Comment (tenisakb) [reply src/components/file-manager/recovery-modal.tsx]: fixed

File: src/components/file-manager/recovery-modal.tsx (hunk 2 above)
Comment (tenisakb) [reply src/components/file-manager/recovery-modal.tsx]: fixed

File: src/components/file-manager/recovery-modal.tsx (hunk 3 above)
Comment (tenisakb) [reply src/components/file-manager/recovery-modal.tsx]: fixed

File: src/components/file-manager/recovery-modal.tsx (hunk 4 above)
Comment (tenisakb) [reply src/components/file-manager/recovery-modal.tsx]: fixed

File: src/components/file-manager/recovery-modal.tsx (hunk 5 above)
Comment (tenisakb) [reply src/components/file-manager/recovery-modal.tsx]: fixed

**Pull request merged**